doc/source/whatsnew/v1.3.0.rst

```diff
@@ -647,6 +647,7 @@ Deprecations
 - Deprecated setting :attr:`Categorical._codes`, create a new :class:`Categorical` with the desired codes instead (:issue:`40606`)
 - Deprecated behavior of :meth:`DatetimeIndex.union` with mixed timezones; in a future version both will be cast to UTC instead of object dtype (:issue:`39328`)
 - Deprecated using ``usecols`` with out of bounds indices for ``read_csv`` with ``engine="c"`` (:issue:`25623`)
+- Deprecated passing arguments (apart from ``labels``) as positional in :meth:`DataFrame.drop` and :meth:`Series.drop` (:issue:`41485`)
```

Comment:
as per other comments, we could consider deprecating some compatibility parameters on Series at the same time.